File: src/main/java/abstract_factory_java/factory/impl/CommonPropertyFactory.java
package abstract_factory_java.factory.impl;

import abstract_factory_java.factory.PropertyFactory;
import abstract_factory_java.property.Car;
import abstract_factory_java.property.House;
import abstract_factory_java.property.Money;
import abstract_factory_java.property.impl.common.CommonCar;
import abstract_factory_java.property.impl.common.CommonHouse;
import abstract_factory_java.property.impl.common.CommonMoney;

/**
 * @author angus
 * @date 2019/11/21
 */
public class CommonPropertyFactory implements PropertyFactory {
    @Override
    public Car buyCar() {
        return new CommonCar();
    }

    @Override
    public House buyHouse() {
        return new CommonHouse();
    }

    @Override
    public Money drawMoney() {
        return new CommonMoney();
    }
}

File: src/main/kotlin/abstract_factory/factory/CommonPropertyFactory.kt
package abstract_factory.factory.impl

import abstract_factory.factory.PropertyFactory
import abstract_factory.property.Car
import abstract_factory.property.House
import abstract_factory.property.Money
import abstract_factory.property.impl.common.CommonCar
import abstract_factory.property.impl.common.CommonHouse
import abstract_factory.property.impl.common.CommonMoney

class CommonPropertyFactory : PropertyFactory {
    override fun buyCar(): Car = CommonCar()

    override fun buyHouse(): House = CommonHouse()

    override fun drawMoney(): Money = CommonMoney()
}